library/x509_crt.c

```diff
@@ -865,25 +867,26 @@ static int x509_crt_parse_der_core( mbedtls_x509_crt *crt, const unsigned char *
         return( MBEDTLS_ERR_X509_INVALID_FORMAT );
     }
 
-    if( len > (size_t) ( end - p ) )
```

Comment:
Didn't we need this check?

Reply:
Sorry, I should have commented on that: ASN.1 parsing function always check that the claimed length doesn't exceed the bounds of the buffer. See [here](https://github.com/ARMmbed/mbedtls/blob/development/library/asn1parse.c#L102).